### PR TITLE
Refactor subroutine explmix in ndrop.F90

### PR DESCRIPTION
**Pull request by brigaudet**
Condense / restructure subroutine explmix in ndrop.F90, add units, and remove unneeded arguments (either explicitly zero or hardwired to zero (flxconv)).

### CODE REVIEW

**Review by tangshuaiqi — APPROVED** [review]

**Review by mingxuanwupnnl — APPROVED** [review]

File: components/eam/src/physics/cam/ndrop.F90 [hunk 1]
```diff
@@ -969,9 +969,10 @@ subroutine dropmixnuc( &
                  + raercol_cw(pver,mm,nsav)*(nact(pver,m) - taumix_internal_pver_inv)
             srcn(pver) = srcn(pver) + max(0.0_r8,tmpa)
          end do
+!! BJG 10/3 remove two arguments of "zero" to explmix here and elsewhere, since have no impact on output.
```
Comment (singhbalwinder) [inline components/eam/src/physics/cam/ndrop.F90]: I think comments regarding code restructuring belong better to Confluence pages unless they are important to understand the code. We can remove this comment.

File: components/eam/src/physics/cam/ndrop.F90 [hunk 2]
```diff
@@ -1168,76 +1169,65 @@ end subroutine dropmixnuc
...
-   ! mixing ratio of ACTIVATED species from previous step
+   real(r8), intent(in),optional :: qactold(pver) 
+   ! number / mass mixing ratio of ACTIVATED species from previous step (# or kg / (kg air)) 
    ! *** this should only be present
    !     if the current species is unactivated number/sfc/mass
 
    integer k,kp1,km1
 
-   if ( is_unact ) then
+!! move if branch inside k loop loop to simplify BJG 9/30
```
Comment (singhbalwinder) [inline components/eam/src/physics/cam/ndrop.F90]: Same here

File: components/eam/src/physics/cam/ndrop.F90 [hunk 3]
```diff
@@ -1168,76 +1169,65 @@ end subroutine dropmixnuc
...
          !        if(q(k)<-1.e-30)then
          !           write(iulog,*)'q=',q(k),' in explmix'
          q(k)=max(q(k),0._r8)
          !        endif
-      end do
+      enddo
 
       !     diffusion loss at base of lowest layer
-      q(pver)=q(pver)-surfrate*qold(pver)*dt+flxconv*dt
+!!  Remove (commented out) line below, since surfrate and flxconv are always zero,
```
Comment (singhbalwinder) [inline components/eam/src/physics/cam/ndrop.F90]: Feel free to remove all these comments for the inactive codes.

**Review by singhbalwinder — CHANGES_REQUESTED** [review]
Looks great! I have some minor comments about removing unused codes and associated comments.

File: components/eam/src/physics/cam/ndrop.F90 [hunk 4]
```diff
@@ -1168,76 +1168,51 @@ end subroutine dropmixnuc
...
+            q(k) = qold(k) + dt*( - src(k) + ekkp(k)*(qold(kp1) - qold(k) +       &
+               qactold(kp1)*(1.0_r8-overlapp(k)))               &
+               + ekkm(k)*(qold(km1) - qold(k) +     &
+               qactold(km1)*(1.0_r8-overlapm(k))) )
+         else
+            q(k) = qold(k) + dt*(src(k) + ekkp(k)*(overlapp(k)*qold(kp1)-qold(k)) +      &
+               ekkm(k)*(overlapm(k)*qold(km1)-qold(k)) )
+         endif
          !        force to non-negative
          !        if(q(k)<-1.e-30)then
```
Comment (singhbalwinder) [inline components/eam/src/physics/cam/ndrop.F90]: get rid of this dead code and the `endif` as well.

**Review by singhbalwinder — APPROVED** [review]
One more minor comment but overall looks great!

**Review by jeff-cohere — APPROVED** [review]

File: components/eam/src/physics/cam/ndrop.F90 (hunk 4 above)
Comment (brigaudet) [reply components/eam/src/physics/cam/ndrop.F90]: OK, done in the latest commit.

**Comment by singhbalwinder** [comment]
All BFB regression tests passed:
```
$ ./cs.status.comp_temp1_explmix_mam4_org_v2_baselines -s
comp_temp1_explmix_mam4_org_v2_baselines: 5 tests
  PASS ERS.ne4pg2_oQU480.F2010.compy_intel
  PASS PEM_P64x1.ne4pg2_oQU480.F2010.compy_intel
  PASS PET.ne4pg2_oQU480.F2010.compy_intel
  PASS SMS_D_Ln5_P32x1.ne4pg2_oQU480.F2010.compy_intel
  PASS SMS_D.ne4pg2_oQU480.F2010.compy_intel
```
Merging it to refactor-maint-2.0

**Pull request merged**